src/status_im/contexts/wallet/send/input_amount/view.cljs

```diff
@@ -199,22 +230,40 @@
          [quo/token-input
           {:container-style style/input-container
            :token           token-symbol
-           :currency        current-currency
-           :crypto-decimals crypto-decimals
+           :currency        fiat-currency
+           :crypto-decimals (min token-decimals 6)
            :error?          (controlled-input/input-error input-state)
            :networks        (seq token-networks)
-           :title           (i18n/label :t/send-limit
-                                        {:limit (make-limit-label current-limit current-currency)})
+           :title           (i18n/label
+                             :t/send-limit
+                             {:limit (if @crypto-currency?
+                                       (make-limit-label-crypto current-limit token-symbol)
+                                       (make-limit-label-fiat current-limit
+                                                              (get constants/currency-symbols
+                                                                   fiat-currency)))})
            :conversion      conversion-rate
            :show-keyboard?  false
-           :value           (controlled-input/input-value input-state)
-           :on-swap         #(reset! crypto-currency? %)
+           :value           input-amount
+           :on-swap         (fn [swap-to-crypto-currency?]
+                              (set-just-toggled-mode? true)
```

Comment:
Currently, when input value changes and the value is valid, routes are fetched. When swapping, the input value changes, the value is valid because it is still the same value in crypto, but routes should not be fetched.

This flag is used to prevent routes being fetched when swapping.